src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -358,20 +387,22 @@ public DomainEntity call() throws Exception {
 
             @Override
             public List<String> call() throws Exception {
-                CriteriaBuilder cb = em.getCriteriaBuilder();
-                CriteriaQuery<DomainEntity> q = cb
+                CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+                CriteriaQuery<DomainEntity> query = criteriaBuilder
                         .createQuery(DomainEntity.class);
-                Root<DomainEntity> r = q.from(DomainEntity.class);
-                q.select(r);
-                List<String> domainList = new ArrayList<String>();
-                List<DomainEntity> result = em.createQuery(q).getResultList();
-                if (result.isEmpty() == false) {
+                Root<DomainEntity> r = query.from(DomainEntity.class);
+                query.select(r);
+                List<String> domainList = null;
+                List<DomainEntity> result = em.createQuery(query).getResultList();
```

Comment:
once again, just return the result list and in this case if nothing is found it should return empty list

Reply:
result is the list of ID's, so cannot directly return the result list. for empty list, yes shall return empty list.

Reply:
got it but you still need to ensure that the returned list is empty if result is empty instead of throwing.